Lib/_collections_abc.py

```diff
@@ -409,6 +415,81 @@ def __subclasshook__(cls, C):
         return NotImplemented
 
 
+class _CallableGenericAlias(GenericAlias):
+    """ Internal class specifically for consistency between the ``__args__`` of
+    ``collections.abc.Callable``'s and ``typing.Callable``'s ``GenericAlias``.
+
+    See :issue:`42195`.
+    """
+
+    def __new__(cls, *args, **kwargs):
+        if not isinstance(args, tuple) or len(args) != 2:
+            raise TypeError("Callable must be used as "
+                            "Callable[[arg, ...], result].")
+        _typ, _args = args
+        if not isinstance(_args, (tuple, EllipsisType)) or len(_args) != 2:
+            raise TypeError("Callable must be used as "
+                            "Callable[[arg, ...], result].")
+        t_args, t_result = _args
+        if not isinstance(t_args, (list, EllipsisType)):
+            raise TypeError("Callable[args, result]: args must be a list. Got"
+                            f" {_type_repr(t_args)}")
+
+        ga_args = []
+        for arg in _args:
+            if isinstance(arg, list):
+                ga_args.extend(arg)
+            else:
+                ga_args.append(arg)
```

Comment:
Since you've already extracted `t_args` and `t_result`, why not do something like
```
if t_args == Ellipsis:
    ga_args = _args
else:
    ga_args = tuple(t_args) + (t_result,)
```
And you won't need the `tuple()` call below.